include/labwc.h

```diff
@@ -80,6 +80,20 @@ struct seat {
 	struct wlr_keyboard_group *keyboard_group;
 
 	bool cursor_requires_fallback;
+	/*
+	 * client_cursor_requested:
+	 *	True if we are expecting a client surface to set its own
+	 *	cursor image.  Set when the client surface is entered.
+	 *	Cleared by cursor_set().
+	 *
+	 * server_cursor_name:
+	 *	Name of most recent server-side cursor image.  Set by
+	 *	cursor_set().  Cleared when a client cursor image is
+	 *	received.
+	 */
+	bool client_cursor_requested;
+	char *server_cursor_name;
```

Comment:
Could we combine these two? Not 100% sure right now but I think `server_cursor_name == NULL` should be the same as `client_cursor_requested`.

Reply:
We could rename the bool to `cursor_set_by_server` though

Reply:
I would like to combine them, didn't really think it through before. I think it will work if we do `zfree(server_cursor_name)` when entering the client surface (in place of `client_cursor_requested = true`).

Reply:
Good point, that should work.